src/coreclr/jit/lowerxarch.cpp

```diff
@@ -2506,8 +2622,65 @@ GenTree* Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
     //          +--*  ... T
     //          +--*  opN T
     //   node = *  HWINTRINSIC   simd   T Create
+    if (intrinsicId == NI_Vector512_Create)
```

Comment:
It'd be nice to merge this code path with the `Vector256` handling just below (not necessary as part of this PR).

If we added a `gtNewSimdWithUpperNode` helper, then we could abstract this away. It would also allow us to make the existing `Vector256.WithUpper` a direct intrinsic rather than requiring inlining and constant folding to kick in.

Reply:
Added a To-Do